api/v1beta1/azureclusteridentity_types.go

```diff
@@ -59,6 +59,9 @@ type AzureClusterIdentitySpec struct {
 	// ClientSecret is a secret reference which should contain either a Service Principal password or certificate secret.
 	// +optional
 	ClientSecret corev1.SecretReference `json:"clientSecret,omitempty"`
+	// certPath is the path where certificates exist. When set, it takes precedence over ClientSecret for types that uses certs like ServicePrincipalCertificate.
```

Comment:
I would replace "uses" with "use" in this sentence (and re-run make generate).

This looks good. Is there somewhere we could add unit test coverage of the case where the optional path is set?

Reply:
Updated and added a unit test. Please let me know if you want me to move that cert file to someplace else. The cert is the same one that is used in the k8s secret path.